src/compiler/checker.ts

```diff
@@ -32376,7 +32375,7 @@ namespace ts {
                 if (getModifierFlags(member) & ModifierFlags.Ambient) {
                     continue;
                 }
-                if (isInstancePropertyWithoutInitializer(member)) {
+                if (isInstancePropertyWithoutInitializer(member) && !isNonNullPropertyWithJSDoc(member)) {

```

Comment:
is this the equivalent of a "definite assignment assertion" on properties?
If so, does this need to be expanded to variables as well?